src/@3rdweb-sdk/react/hooks/useEngine.ts

```diff
@@ -1270,3 +1281,36 @@ export function useEngineSubscriptionsLastBlock(
     },
   );
 }
+
+export interface EngineResourceMetrics {
+  error: string;
+  data: {
+    cpu: number;
+    memory: number;
+  };
+}
+
+export function useEngineResourceMetrics(instanceUrl: string) {
```

Comment:
Let's think about piping engineId in here. I want to avoid passing domains/slugs as identifiers whenever possible.